neuraxle/pipeline.py

```diff
@@ -62,79 +34,158 @@ def __hash__(self):
+
+        return new_self
+
+    def fit_transform_steps(self, data_inputs, expected_outputs):
+        """
+        After loading the last checkpoint, fit transform each pipeline steps
+        :param data_inputs: the data input to fit on
+        :param expected_outputs: the expected data output to fit on
+        :return: the pipeline itself
+        """
+        steps_left_to_do, data_inputs = self.load_checkpoint(data_inputs)
+
+        new_steps_as_tuple: NamedTupleList = []
+        for step_name, step in steps_left_to_do:
+            step, data_inputs = step.fit_transform(data_inputs, expected_outputs)
+            new_steps_as_tuple.append((step_name, step))
+
+        self.steps_as_tuple = self.steps_as_tuple[:len(self.steps_as_tuple) - len(steps_left_to_do)] + \
+                              new_steps_as_tuple
+
+        return self, data_inputs
+
+    def transform(self, data_inputs):
+        """
+        After loading the last checkpoint, transform each pipeline steps
+        :param data_inputs: the data input to fit on
+        :return: transformed data inputs
+        """
+        steps_left_to_do, data_inputs = self.load_checkpoint(data_inputs)
+        for step_name, step in steps_left_to_do:
+            data_inputs = step.transform(data_inputs)
+
+        return data_inputs
+
+    def inverse_transform_processed_outputs(self, processed_outputs) -> Any:
+        """
+        After transforming all data inputs, and obtaining a prediction, we can inverse transform the processed outputs
+        :param processed_outputs: the forward transformed data input
+        :return: backward transformed processed outputs
+        """
+        for step_name, step in list(reversed(self.steps_as_tuple)):
+            processed_outputs = step.transform(processed_outputs)
+        return processed_outputs
 
-        reversed_steps_as_tuple = list(reversed(self.steps_as_tuple))
-        return self.pipeline_runner.set_steps(reversed_steps_as_tuple).transform(processed_outputs)
+    def load_checkpoint(self, data_inputs, expected_outputs=None):
+        """
+        Find the starting step index, and its corresponding data inputs using the checkpoint steps.
+        If the checkpoint is inside another step, the starting step will be step that contains the checkpoint
+        :param expected_outputs: expected outputs to fit on
+        :param data_inputs: the data input to fit on
+        :return: index, (data_inputs, expected_outputs) tuple for the starting step data inputs-outputs
+         and the starting step index
+        """
+        new_data_inputs = data_inputs
+        new_starting_step_index = 0
+        found_checkpoint = False
+
+        for index, (step_name, step) in enumerate(reversed(self.steps_as_tuple)):
+            if not found_checkpoint and isinstance(step, ResumableStep) and step.should_resume(data_inputs):
+                _, checkpoint = step.load_checkpoint(data_inputs)
```

Comment:
Call read_checkpoint here directly and delete load_checkpoint from the ResumableStep's interface.

Reply:
I cannot delete load checkpoint from the ResumableSteps interface because Pipeline has to have it. Pipeline and BaseCheckpointStep need to implement this method for the algorithm to work. I renamed load_checkpoint to read_checkpoint in the interface to have the same name.

Reply:
nevermind read the others comment. I get it now